app/com/linkedin/drelephant/spark/heuristics/StagesAnalyzer.scala

```diff
@@ -0,0 +1,479 @@
+        details)
+
+      val (rawSkewSeverity, taskSkewSeverity) =
+        checkForTaskSkew(stageId, stageData, medianTime, maxTime, stageDuration, details,
+          executionSpillSeverity)
+
+      val failedTasks = failedTasksStageMap.get(stageId)
+
+      val (numTasksWithOOM, failedWithOOMSeverity) =
+        checkForTaskError(stageId, stageData, failedTasks,
+          StagesWithFailedTasksHeuristic.OOM_ERROR, "of OutOfMemory exception.",
+          details)
+
+      val (numTasksWithContainerKilled, failedWithContainerKilledSeverity) =
+      checkForTaskError(stageId, stageData, failedTasks,
+        StagesWithFailedTasksHeuristic.OVERHEAD_MEMORY_ERROR,
+        "the container was killed by YARN for exeeding memory limits.", details)
+
+      val taskFailureSeverity = calculateTaskErrorSeverity(stageData.numFailedTasks,
+        stageData.numTasks)
+
+      val stageFailureSeverity = if (stageData.status == StageStatus.FAILED) {
+        val reason = stageData.failureReason.getOrElse("")
+        details += s"Stage $stageId failed: $reason"
+        Severity.CRITICAL
+      } else {
+        Severity.NONE
+      }
+
+      val gcSeverity = checkForGC(stageId, stageData)
+
+      new StageAnalysis(stageData.stageId, rawSpillSeverity, executionSpillSeverity,
+        longTaskSeverity, rawSkewSeverity, taskSkewSeverity, failedWithOOMSeverity,
+        failedWithContainerKilledSeverity, taskFailureSeverity, stageFailureSeverity,
+        gcSeverity, stageData.numTasks, medianTime, maxTime, stageData.memoryBytesSpilled,
+        maxTaskSpill, stageData.inputBytes, stageData.numFailedTasks,numTasksWithOOM,
+        numTasksWithContainerKilled, stageDuration, details)
+    }
+  }
+
+  /**
+    * Check stage for execution memory spill.
+    *
+    * @param stageId stage ID.
+    * @param stageData stage data.
+    * @param details (IN/OUT) information and recommendations -- any new recommendations
+    *                from analyzing the stage for execution memory spill will be appended.
+    * @return the severity for just execution memory spill, and the severity to report
+    *         (warning is suppressed if processing more than DEFAULT_MAX_DATA_PROCESSED_THRESHOLD
+    *         data, since some spill is acceptable/unavoidable when processing large amounts
+    *         of data).
+    */
+  private def checkForExecutionMemorySpill(
+      stageId: Int,
+      stageData: StageData,
+      details: ArrayBuffer[String]): (Severity, Severity) = {
+    val maxData = Seq(stageData.inputBytes, stageData.shuffleReadBytes,
+      stageData.shuffleWriteBytes, stageData.outputBytes).max
+    val rawSpillSeverity = executionMemorySpillThresholds.severityOf(
+      stageData.memoryBytesSpilled / maxData.toDouble)
+    val tmp = DEFAULT_MAX_DATA_PROCESSED_THRESHOLD
+    val executionSpillSeverity = if (maxData < DEFAULT_MAX_DATA_PROCESSED_THRESHOLD) {
+      rawSpillSeverity
```

Comment:
It seems this condition should be (maxData > DEFAULT_MAX_DATA_PROCESSED_THRESHOLD)